pkg/telemetry/tracing.go

```diff
@@ -40,6 +41,13 @@ func MustNewTracerProvider(endpoint string, serviceName string, ratio float64) *
 		panic(fmt.Sprintf("failed to establish a connection with the otlp exporter: %v", err))
 	}
 
+	if enableTailLatencySpanExporter {
+		if ratio != 1 {
```

Comment:
Why require this? If you want to downsample AND only export high latency spans, then I don't see a problem with that.

I'd avoid being over pedantic here on this check.